streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -119,6 +122,24 @@ public void configure(Map<String, ?> configs) {
         streamThread = (StreamThread) o;
         streamThread.partitionAssignor(this);
 
+        String userEndPoint = (String) configs.get(StreamsConfig.USER_ENDPOINT_CONFIG);
+        if (userEndPoint != null && !userEndPoint.isEmpty()) {
+            final String[] hostPort = userEndPoint.split(":");
+            if (hostPort.length != 2) {
+                log.warn("Config '{}' isn't in the correct host:pair format. This instance will not be available for discovery"
```

Comment:
I replied to the KIP-67 discussion thread with a comment regarding the current `host:port` approach.  In a nutshell, I am leaning towards relaxing the requirement (that it must be a `host:port` pattern).  For the same reason I'd not throw an exception here.  In fact, we may not even want to log a warning if we actually do decide to not enforce any `host:port` pattern.